### DISCUSSION TITLE
第2节课笔记(15班)

### DISCUSSION THREAD

**Discussion by ironartisan** [InternLM/Tutorial#37] [笔记]
请在下方提交笔记链接

**Comment by Mrguanglei** [comment]
nice，非常棒的训练营，以前是个小白完全不会去如何部署，和编写代码去下载模型，虽然现在还不会如何训练但是我感觉我已经前进一小步啦，希望未来完全能够学会大模型。加油
飞书笔记：https://gzr5xfkph4.feishu.cn/docx/I3KPdL7Auo65M6x0klccfP2Fnoh?from=from_copylink

**Reply by Mrguanglei** [reply]
https://gzr5xfkph4.feishu.cn/docx/I3KPdL7Auo65M6x0klccfP2Fnoh?from=from_copylink

**Comment by ziptoam** [comment]
第二节课比较有意思，是跟着教程一步一步的跑通三个已经预设好的模型。
1.lagent模型：留下的疑问是，为什么有一些计算题，计算不准确，影响准确的因素是什么
2.浦语灵笔：留下的疑问是，生成文案中的图片，是在图片库中引用的，还是说和文本一样，也是生成的；另外，图片内容理解，我上传了一张小女孩在公园荡秋千的照片，模型没能够理解图片中有小女孩和秋千，只有空荡荡的公园，不知道图片内容理解准确性会受哪些因素影响？

**Comment by szhowardhuang** [comment]
这节课按照讲义做很顺，没什么问题。
以下记录一下留意到的。

1）x-composer环境安装有个error,如下：

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
triton 2.0.0 requires cmake, which is not installed.
triton 2.0.0 requires lit, which is not installed.

不过好像不影响demo


2）SSH体会：internLLM的开发机用webui，需要本地端口转发，怎么实现呢？

在本地终端输入以下指令，第一个6006是在本地打开的端口，127.0.0.1:6006则是在开发机（服务器）打开的端口，而33386是开发机的SSH端口

ssh -CNg -L 6006:127.0.0.1:6006 root@ssh.intern-ai.org.cn -p 33386


含义有3点： 
服务器上的 6006 端口服务映射到本地 6006
本地 6006 的网络数据转发到 服务器 的 6006 端口
本地通过SSH端口33386和服务器建立SSH连接


参数说明：
-C: 压缩传输，提高传输速度。
-N: 建立静默连接（建立了连接但看不到具体会话）
-g: 在 -L 参数中，允许远程主机连接到建立的转发端口。
-L: 本地端口转发
-p: 指定 SSH 端口


查看.ssh目录的config文件，如下：

Host ssh.intern-ai.org.cn
  HostName ssh.intern-ai.org.cn
  Compression yes
  GatewayPorts yes
  LocalForward 6006:127.0.0.1 6006
  User root
  Port 33386
  StrictHostKeyChecking no


此时6006端口建立完成，但是没有web服务在上面，
当我们run webui的时候，比如：
streamlit run /root/code/lagent/examples/react_web_demo.py --server.address 127.0.0.1 --server.port 6006

我们指定了webui在127.0.0.1：6006 端口提供服务，于是开始进行webui通讯。

**Comment by dqyydq** [comment]
https://blog.csdn.net/qq_52410386/article/details/135444418

**Comment by lizhunkg** [comment]
第二课，笔记作业：https://note.youdao.com/s/PjkZ2Elf

**Comment by IyGnahz** [comment]
笔记&作业：https://g6xuom3zbh.feishu.cn/docx/VwKGd08m0oxbzvxPJRScMtPhnwb

**Comment by scchy** [comment]
第二课笔记
https://github.com/scchy/My_Learn/blob/master/LLM_prime/2_InternLM_demo_doc.md

**Comment by yabgdsajghsj** [comment]
[Uploading 第二课笔记.pdf…]()
[Uploading 第二课笔记.pdf…]()

**Comment by pyajagod** [comment]
第二节课内容是，简单使用internLM的几个模型实践了有意思的Demo。提供的资料非常完善，所以几个Demo完成的也非常顺利。所以记录一下遇到的问题以及解决方案。
是在本地运行web-demo时候，使用ssh -CNg -L 6006:127.0.0.1:6006 root@ssh.intern-ai.org.cn -p 39999命令，可能会遇到如下问题

`@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ED25519 key sent by the remote host is
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.
Please contact your system administrator.
Add correct host key in ～/.ssh/known_hosts to get rid of this message.
Offending ECDSA key in ～/.ssh/known_hosts:6
Host key for [ssh.intern-ai.org.cn]:34014 has changed and you have requested strict checking.
Host key verification failed.`
解决方法是到～/.ssh/known_hosts文件中，找到与internStdio相关的几条记录，删掉后再运行命令就可以啦。

**Comment by bblgbr** [comment]
第二次课笔记：02 demo尝试.md
https://github.com/bblgbr/play_llm/blob/main/InterLM_Class/02%20Demo%E5%B0%9D%E8%AF%95.md

**Comment by 6forwater29** [comment]
非常丝滑，非常好

**Comment by 6forwater29** [comment]
非常丝滑，非常好

**Comment by woodx9** [comment]
包括一些知识点ppt，和实操上的qa笔记 [飞书链接](https://gxloh9mfvxh.feishu.cn/wiki/O21zwwNL5iq7KrkEFxwcNGDAnVf)

**Comment by 520Enterprise** [comment]
笔记&作业：https://www.cnblogs.com/520Enterprise/p/17945399

**Comment by Visytudz** [comment]
#### Problem
无法直接通过Huggingface下载模型，可换源为[Huggingface镜像站](https://hf-mirror.com/)。导入环境变量`export HF_ENDPOINT=https://hf-mirror.com`后可正常下载。

```python
import os
from huggingface_hub import hf_hub_download  # Load model directly
hf_hub_download(repo_id="internlm/internlm-7b", filename="config.json")
```

**Comment by sokolo05** [comment]
https://github.com/sokolo05/Scholar_PuYu/blob/main/02.%E8%AF%BE%E7%A8%8B%E7%AC%94%E8%AE%B0/%E7%AC%AC%E4%BA%8C%E8%8A%82_%E7%AC%94%E8%AE%B0.md

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by Gaia-yue** [comment]
https://github.com/Gaia-yue/---/blob/main/%E4%B8%80%E6%9C%88/2_Demo.md

**Comment by seienn** [comment]
https://gvxxiiwsufy.feishu.cn/wiki/BqCcw4Nntin1u1k3Um0cee1Bnde?from=from_copylink

**Comment by NANDSS1** [comment]
https://www.yuque.com/nandss/gyyg9p/ok7ng4msgbipyk1u?singleDoc# 《lesson2 书生·浦语大模型Demo笔记》

**Comment by g29times** [comment]
https://novagen.notion.site/2-94537b2263154cd6823ae35a9b345189?pvs=4
